.github/workflows/ubuntu.yml

```
@@ -0,0 +1,84 @@
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: true
+      matrix:
+        os: [ubuntu-24.04]
+        build_type: [Release, Debug]
+        cpp_compiler: [g++, clang++]
+
+    steps:
+    - uses: actions/checkout@v3
```

actionlint (1.7.4)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)